nulib/dc-api-v2
# Log dev team flag to metrics

### Summary

We want to be able to sort out our own requests and testing of DC chat through metrics

### Changelog

- Adds `isDevTeam` flag to DCAPI token
- Add `is_dev_team` flag to the metrics log
- Also fixes issue where Meadow's superuser tokens where not authorized for chat because they are `isLoggedIn=false`

### Steps to Test

- Grab new tfvars: https://github.com/nulib/tfvars/blob/main/dc-api/staging.parameters#L11
- Update any dev params accordingly
- dev/deploy chat 
- Run main DCAPI dev/locally
- Generate token with the `isDevTeam` flag set
  - You can do this by hitting https://{your-dev-env}.dev.rdc.library.northwestern.edu:3002/auth/login?goto=https://<your-dev-env>{your-dev-env}.dev.rdc.library.northwestern.edu:3002/auth/callback (and grabbing the resulting token)
- Check the `auth/whoami` route that the token contains the flag
- Send a chat request (through Postman)
- check the metrics log (compare with screenshot below)
- Test other types of users who should result in a false dev team flag


![Screenshot 2024-08-29 at 9 39 03 AM](https://github.com/user-attachments/assets/1955d87e-ddf3-48ad-ae50-5875bd27ffd6)

![Screenshot 2024-08-29 at 9 40 30 AM](https://github.com/user-attachments/assets/ca0cb537-5fd5-4f00-9bc2-dce5c8e2b893)
